lang/funcs/core/net/is_macaddress_func.go

```diff
@@ -0,0 +1,94 @@
+// Mgmt
+// Copyright (C) 2013-2021+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package corenet
+
+import (
+	"strings"
+
+	"github.com/purpleidea/mgmt/lang/funcs/simple"
+	"github.com/purpleidea/mgmt/lang/types"
+)
+
+func init() {
+	simple.ModuleRegister(ModuleName, "ismacaddress", &types.FuncValue{
+		T: types.NewType("func(a str) bool"),
+		V: IsMacAddress,
+	})
+}
+
+// 00:21:cc:be:e0:6b
+
+// IsMacAddress verifies that input string is a mac address
+func IsMacAddress(input []types.Value) (types.Value, error) {
+	mac := input[0].Str()
+
+	if len(mac) != 17 {
+		return &types.BoolValue{V: false}, nil
+	}
+	delimiter := strings.Split(mac, "")[2]
+	return &types.BoolValue{V: validate(mac, delimiter)}, nil
+}
+
+func validate(mac, delimiter string) bool {
+	// valid delimiters
+	delims := map[string]bool{
+		":": true,
+		"-": true,
+		".": true,
+	}
+	if !delims[delimiter] {
```

Comment:
I tried that out, if delimiter is not in the map it will be false and since the logic statement its asking for the reverse of that, it will be true, once inside the statement it'll then return false as it should because "X" isn't one of the accepted delimiters, and the mac address string will be invalid as well. I'm not sure on a safe way to do this though.

Reply:
Here's an example:

https://play.golang.org/p/txidCnVqfai

For other maps, this can be a panic depending on what the value is. In this case it's a bool, so it's safe. But that's accidental.

Reply:
ok I see, when assigning a map[key] to a variable go also returns a bool if it exists and you can also assign that to a variable.